
# seuratter

<!-- badges: start -->
<!-- badges: end -->

The goal of seuratter is to assist the Single Cell RNA Sequencing workflow carried out by Seurat and
to provide a different perspective to the data analysis component through generation of a graph.

## Installation

You can install the released version of seuratter like:

``` r
require("devtools")
install_github("sotaro0214/seuratter")
library("seuratter")
```

## Overview

An overview of the package is illustrated in the image below.

![](./inst/extdata/SILVA_A_A1.png) 

## Contributions

The author of this package is Sotaro Hirai. The function available within this package include:

``` r
library("seuratter")
lsf.str("package:seuratter")
```

- determine_dimensionality
- compare_cluster_genes
- extract_genes_expressed
- draw_cluster_gene_relations

The above 4 functions were developed by Sotaro. The contributions of Sotaro come from two aspects of this package.
The other contributions come from the packages Seurat, dplyr, and igraph, which Seurat and dplyr are used for carrying out the standard single cell rna seq workflow to generate relevant data and igraph is used for producing a graphical
output in the end.
One of Sotaro's contributions comes from the function determine_dimensionality. The function determine_dimensionality takes as an input an elbow plot and determines an 'elbow', a point at which the slope begins to relax. This method is 
usually carried out by humans and is quite arbitrary, so this function aims to automate this procedure so that this determination becomes less arbitrary.
Sotaro's other contribution comes from the latter 3 functions. These are used in combination with the igraph package to generate a graph from the data generated by Seurat. One analysis Seurat lacks is the aspect of determining how the clustered cells relate to cells from other clusters. extract_genes_expressed takes in data from Seurat and is used within compare_cluster_genes. draw_cluster_gene_relations then uses the output to generate a graph that maps the relationships between clusters, based on the similarity in gene expression profiles. The more related they are, the more stronger the connections are represented. Seurat allows multiple clusters to cluster into a larger single cluster by using UMAP, but seuratter's approach allows to portray how these clusters within a higher-level cluster relate to each other. Some of the clusters within one high-level cluster may be more related to each other than others, and determining this allows us to observe the data at the same level of abstractness as Seurat, but at a higher resolution/definition.

